assets/js/components/Button.js

```diff
@@ -21,72 +21,69 @@
  */
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
+import useMergedRef from '@react-hook/merged-ref';
+
 /**
  * WordPress dependencies
  */
-import { Component, createRef } from '@wordpress/element';
+import { forwardRef, useRef, useEffect } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import { MDCRipple } from '../material-components';
 
-class Button extends Component {
-	constructor( props ) {
-		super( props );
-		this.buttonRef = createRef();
-	}
-
-	componentDidMount() {
-		new MDCRipple( this.buttonRef.current );
-	}
+const Button = forwardRef( ( {
+	children,
+	href,
+	text,
+	className,
+	danger,
+	disabled,
+	target,
+	icon,
+	trailingIcon,
+	ariaHaspopup,
+	ariaExpanded,
+	ariaControls,
+	...extraProps
+}, ref ) => {
+	const buttonRef = useRef( null );
+	useEffect( () => {
+		MDCRipple.attachTo( buttonRef.current );
```

Comment:
@aaemnnosttv I did some digging and the ref is set after the first render(), but before componentDidMount() in the component lifecycle as per this article: https://hackernoon.com/refs-in-react-all-you-need-to-know-fb9c9e2aeb81
I'am not sure we should add an unnecessary check for that. Besides, in every examples I've looked, and the official docs, there is no check for the current value.

Reply:
I created a sandbox to test this here: https://codesandbox.io/s/bold-mcclintock-u2w71?file=/src/App.js so this should work.

The docs reference [using a callback style ref](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node) for the case where you want to "subscribe" to changes to the current value.

> We didn’t choose useRef in this example because an object ref doesn’t notify us about changes to the current ref value. Using a callback ref ensures that even if a child component displays the measured node later (e.g. in response to a click), we still get notified about it in the parent component and can update the measurements.

When putting `someRef.current` in a dependency array, it actually raises an ESLint rule, if using `react-hooks/exhaustive-deps`:

> React Hook useEffect has an unnecessary dependency: 'appRef.current'. Either exclude it or remove the dependency array. Mutable values like 'appRef.current' aren't valid dependencies because mutating them doesn't re-render the component.

In most cases, I don't think this will make much of a difference since the inner element is always the same. In this case however, the component may render an anchor or a button element depending on props. Using the current value in the effect dependency array does trigger the effect again as expected though so not sure there is any real benefit in changing the usage here. @tofumatt ?

Reply:
Looks like removing the dependency array entirely works fine and makes sense to me. I suppose it results in `MDCRipple.attachTo` being called even if `buttonRef.current` hasn't changed, but that's fine I think?

I do feel weird not protecting against `buttonRef.current` being `undefined`/`null`, as in this case it's fine but I thought it some it wouldn't be set. It's just a bit confusing having to figure out when it's necessary depending on the component. I usually just always do it for a "belt and braces" approach, but fair enough if it's not needed here.

Removing the dependency array should be fine here from testing in the above sandbox.

Reply:
I'm not sure if calling `MDCRipple.attachTo` without an element is okay or not. I dug into the components there a bit but couldn't really find an answer to that so I just tried it in a test: it errors.

![image](https://user-images.githubusercontent.com/1621608/99795869-2b05e600-2b35-11eb-95bb-a0f29cd22670.png)

The `attachTo` method is part of the base mdc component so this would apply to all `MDC*` classes, not just Ripple.

So in that case, I think we just need to add a conditional around this call to make sure `current` is not empty – I don't think we need to check specifically that it is an instance of `Element`.